extensions/amp-font/0.1/amp-font.js

```diff
@@ -0,0 +1,176 @@
+/**
+ * Copyright 2015 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+/**
+ * @fileoverview Triggers and monitors loading of custom fonts on AMP pages.
+ * Example:
+ * <code>
+ * <amp-font
+ *   font-family="My Font"
+ *   timeout="3s"
+ *   font-weight="bold"
+ *   on-error-add-class="myfont2-missing"
+ *   on-error-remove-class="myfont3-loaded"
+ *   on-load-add-class="myfont2-loaded"
+ *   on-load-remove-class="myfont1-loaded"
+ *   layout="nodisplay">
+ * </amp-font>
+ * </code>
+ *
+ * the amp-font element's layout type is nodisplay.
+ */
+
+import {FontLoader} from './fontloader';
+import {removeElement} from '../../../src/dom';
+import {timer} from '../../../src/timer';
+import {vsync} from '../../../src/vsync';
```

Comment:
We have to expose vsync via window-based service. We can no longer use it here as a singleton otherwise it will de-factor break mutations queue.